packages/grafana-ui/src/components/Graph/Graph.tsx

```diff
@@ -179,6 +202,70 @@ export class Graph extends PureComponent<GraphProps, GraphState> {
     });
   };
 
+  renderContext = () => {
+    const { series } = this.props;
+    const { contextPos, contextItem, isContextVisible } = this.state;
+
+    if (!isContextVisible || !contextPos || !contextItem || series.length === 0) {
+      return null;
+    }
+
+    // Indicates column(field) index in y-axis dimension
+    const seriesIndex = contextItem ? contextItem.series.seriesIndex : 0;
+    // Indicates row index in context field values
+    const rowIndex = contextItem ? contextItem.dataIndex : undefined;
+
+    const contextDimensions: ContextDimensions<GraphDimensions> = {
+      // Described x-axis context item
+      xAxis: [seriesIndex, rowIndex],
+      // Describes y-axis context item
+      yAxis: contextItem ? [contextItem.series.seriesIndex, contextItem.dataIndex] : null,
+    };
+
+    const dimensions: GraphDimensions = {
+      // time/value dimension columns are index-aligned - see getGraphSeriesModel
+      xAxis: createDimension(
+        'xAxis',
+        series.map(s => s.timeField)
+      ),
+      yAxis: createDimension(
+        'yAxis',
+        series.map(s => s.valueField)
+      ),
+    };
+
+    const formatDate = (date: DateTimeInput, format?: string) => {
+      return dateTime(date)?.format(format);
+    };
+
+    const closeContext = () => this.setState({ isContextVisible: false });
+
+    const getContextMenuSource = () => {
+      return {
+        datapoint: contextItem.datapoint,
+        dataIndex: contextItem.dataIndex,
+        series: contextItem.series,
+        seriesIndex: contextItem.series.seriesIndex,
+        pageX: contextPos.pageX,
+        pageY: contextPos.pageY,
+      };
+    };
+
+    const contextContentProps: GraphContextMenuProps = {
+      x: contextPos.pageX,
+      y: contextPos.pageY,
+      onClose: closeContext,
+      getContextMenuSource: getContextMenuSource,
+      formatSourceDate: formatDate,
+      dimensions,
+      contextDimensions,
+    };
+
+    const contextContentRender = GraphContextMenu;
+    const contextContent = React.createElement(contextContentRender, { ...contextContentProps });
```

Comment:
Is there a reason to use `React.createElement` instead of `< GraphContextMenu {...props}/>`?

Reply:
Not really. Again, I was trying to be consistent/follow the same style as `renderTooltip()`. In `renderTooltip()` there is a reason for using `React.createElement`, but here is not a particular reason for that. 

Would `< GraphContextMenu {...props}/>` be a preferred way?

Reply:
> Would < GraphContextMenu {...props}/> be a preferred way?

I think so if we do not need it. Otherwise it just makes you think why `createElement` and whether there is something special that you do not see.

Reply:
For tooltip there is actually a reason for doing that. When you do 
```
<Graph ... >
  <Chart.Tooltip ... />
</Graph>
```
It doesn't mean that the tooltip is render. It only indicates, that this visualisation wants to use the tooltip. Then it's up to the vis component to decide when and how(contolled via tooltipComponent prop). This is a pattern we want to introduce but we need some more time to establish it.

Reply:
@dprokop That is because you do not have to render children, but you could change the line https://github.com/grafana/grafana/blob/0ebfd921cfd317b0eeef112719dbca063920db99/packages/grafana-ui/src/components/Graph/Graph.tsx#L167 to `const tooltipContent = <TooltipContentRenderer { ...tooltipContentProps }/>;` as far as I know those two expressions should be equivalent as one is transformed to other.